ui/src/store.js

```diff
@@ -0,0 +1,186 @@
+import autodux from 'autodux';
+import { doFetch } from './utils/fetch-websocket';
+
+export const {
+  reducer,
+  initial: defaultState,
+  actions: {
+    activateConnection,
+    deactivateConnection,
+    serverConnected,
+    serverDisconnected,
+    updatePurses,
+    changePurse,
+    swapInputs,
+    changeAmount,
+    createOffer,
+    resetState,
+    updateInvitationDepositId,
+  },
+} = autodux({
+  slice: 'autoswap',
+  initial: {
+    active: false,
+    connected: false,
+    account: null,
+    purses: null,
+    inputPurse: null,
+    outputPurse: null,
+    inputAmount: null,
+    outputAmount: null,
+    freeVariable: null,
+    invitationDepositId: null,
+  },
+  actions: {
+    activateConnection: state => ({ ...state, active: true }),
```

Comment:
These are the same actions as above. Do they need to be defined in both places?  What if they diverge?

Reply:
They are destructured exports in the first case and parameters to autodux in the second case.  If they diverge, then the value of the exported symbol will be `undefined`.  This WET matching-up of a value with its binding is still better than having to match up 3 values with a binding, but I agree DRY would be nicer.